paasta_tools/remote_git.py

```diff
@@ -76,6 +78,7 @@ class LSRemoteException(Exception):
     pass
 
 
+@timeout(error_message="Timed out connecting to git server, is it reachable from where you are?", use_signals=False)
```

Comment:
can you be explicit about the time seconds value here?

I'm guessing it is 10. Running this on yelp-main takes about 3. 20seconds might be a sane compromise.